ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -132,6 +133,93 @@ export class AnsClient {
     return pendingTransaction.hash;
   }
 
+  /**
+   * Mint a new Aptos Subdomain
+   *
+   * @param account AptosAccount the owner of the domain name
+   * @param subdomainName subdomain name to mint
+   * @param domainName Aptos domain name to mint under
+   * @param expirationTimestamp must be set between the domains expiration and the current time
+   * @returns The hash of the pending transaction submitted to the API
+   */
+  async mintAptosSubdomain(
+    account: AptosAccount,
+    subdomainName: string,
+    domainName: string,
+    expirationTimestamp: number,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<Gen.HashValue> {
+    // check if the name is valid
+    if (domainName.match(nameComponentPattern) === null) {
+      throw new ApiError(400, `Name ${domainName} is not valid`);
+    }
+    // check if the name is valid
+    if (subdomainName.match(nameComponentPattern) === null) {
+      throw new ApiError(400, `Name ${subdomainName} is not valid`);
+    }
```

Comment:
Should we say `Domain name` and `Subdomain name`

Reply:
will fix